src/test/java/edu/ksu/canvas/attendance/submitter/CanvasAssignmentAssistantUTest.java

```diff
@@ -0,0 +1,176 @@
+
+    private CanvasAssignmentAssistant canvasAssignmentAssistant;
+
+    @Mock
+    private AttendanceAssignmentService assignmentService;
+
+    @Mock
+    private AttendanceSectionService attendanceSectionService;
+
+    @Mock
+    private AttendanceAssignmentRepository assignmentRepository;
+
+    @Mock
+    private CanvasApiWrapperService canvasApiWrapperService;
+
+    private NonRefreshableOauthToken oauthToken;
+    private AttendanceAssignment attendanceAssignment;
+    private Optional<Assignment> assignmentOptional;
+    private Assignment assignment;
+    private AttendanceSection attendanceSection;
+    private List<AttendanceSection> sectionList;
+
+    @Before
+    public void setup() {
+
+        canvasAssignmentAssistant = new CanvasAssignmentAssistant();
+
+        oauthToken = new NonRefreshableOauthToken(OAUTH_STRING);
+
+        attendanceAssignment = new AttendanceAssignment();
+        attendanceAssignment.setAssignmentId(ASSIGNMENT_ID);
+        attendanceAssignment.setAssignmentName(ASSIGNMENT_NAME);
+        attendanceAssignment.setCanvasAssignmentId(CANVAS_ASSIGNMENT_ID);
+        attendanceAssignment.setGradingOn(true);
+        attendanceAssignment.setAssignmentPoints(ASSIGNMENT_POINTS);
+        attendanceAssignment.setAbsentPoints(0.0);
+        attendanceAssignment.setExcusedPoints(0.0);
+        attendanceAssignment.setTardyPoints(0.0);
+        attendanceAssignment.setPresentPoints(100.0);
+
+        assignment = new Assignment();
+        assignment.setId(Long.toString(ASSIGNMENT_ID));
+        assignment.setName(attendanceAssignment.getAssignmentName());
+        assignment.setPointsPossible(Double.valueOf(attendanceAssignment.getAssignmentPoints()));
+        assignment.setCourseId(Long.toString(COURSE_ID));
+        assignment.setDescription(ASSIGNMENT_DESCRIPTION);
+        assignment.setPublished(true);
+        assignment.setUnpublishable(false);
+        assignmentOptional = Optional.of(assignment);
+
+        attendanceSection = new AttendanceSection();
+        attendanceSection.setCanvasCourseId(COURSE_ID);
+        attendanceSection.setSectionId(SECTION_ID);
+        attendanceSection.setName(SECTION_NAME);
+        attendanceSection.setCanvasSectionId(SECTION_ID);
+        sectionList = new ArrayList<>();
+        sectionList.add(attendanceSection);
+        attendanceAssignment.setAttendanceSection(attendanceSection);
+
+        Whitebox.setInternalState(canvasAssignmentAssistant, "assignmentService", assignmentService);
+        Whitebox.setInternalState(canvasAssignmentAssistant, "attendanceSectionService", attendanceSectionService);
+        Whitebox.setInternalState(canvasAssignmentAssistant, "assignmentRepository", assignmentRepository);
+        Whitebox.setInternalState(canvasAssignmentAssistant, "canvasApiWrapperService", canvasApiWrapperService);
+    }
+
+    @Test
+    public void createAssignmentInCanvasHappyPath() throws IOException, AttendanceAssignmentException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.io.IOException, edu.ksu.canvas.attendance.exception.AttendanceAssignmentException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1160)